simul/platform/mininet/start.py

```diff
@@ -0,0 +1,243 @@
+        super(BaseRouter, self).terminate()
+
+
+class Cothority(Host):
+    """A cothority running in a host"""
+    def config(self, gw=None, simul="", **params):
+        self.gw = gw
+        self.simul = simul
+        super(Cothority, self).config(**params)
+        if runSSHD:
+            self.cmd('/usr/sbin/sshd -D &')
+
+    def startCothority(self):
+        socat="socat -v - udp-sendto:%s:%d" % (self.gw, socatPort)
+
+        args = "-debug %s -address %s:2000 -simul %s" % (debugging, self.IP(), self.simul)
+        if True:
+            args += " -monitor %s:10000" % global_root
+        ldone = ""
+        if self.IP().endswith(".0.2"):
+            ldone = "; date > " + logdone
+        dbg( 3, "Starting cothority on node", self.IP(), ldone )
+        self.cmd('( ./cothority %s 2>&1 %s ) | %s &' %
+                     (args, ldone, socat ))
+
+    def terminate(self):
+        dbg( 3, "Stopping cothority" )
+        self.cmd('killall socat cothority')
+        super(Cothority, self).terminate()
+
+
+class InternetTopo(Topo):
+        """Create one switch with all hosts connected to it and host
+        .1 as router - all in subnet 10.x.0.0/16"""
+        def __init__(self, myNet=None, rootLog=None, **opts):
+            Topo.__init__(self, **opts)
+            server, mn, n = myNet[0]
+            switch = self.addSwitch('s0')
+            baseIp, prefix = netParse(mn)
+            gw = ipAdd(1, prefix, baseIp)
+            dbg( 2, "Gw", gw, "baseIp", baseIp, prefix,
+                 "Bandwidth:", bandwidth, "- delay:", delay)
+            hostgw = self.addNode('h0', cls=BaseRouter,
+                                  ip='%s/%d' % (gw, prefix),
+                                  inNamespace=False,
+                                  rootLog=rootLog)
+            self.addLink(switch, hostgw)
+
+            for i in range(1, int(n) + 1):
+                ipStr = ipAdd(i + 1, prefix, baseIp)
+                host = self.addHost('h%d' % i, cls=Cothority,
+                                    ip = '%s/%d' % (ipStr, prefix),
+                                    defaultRoute='via %s' % gw,
+			                	    simul=simulation, gw=gw)
+                dbg( 3, "Adding link", host, switch )
+                self.addLink(host, switch, bw=bandwidth, delay=delay)
+
+def RunNet():
+    """RunNet will start the mininet and add the routes to the other
+    mininet-services"""
+    rootLog = None
+    if myNet[1] > 0:
+        i, p = netParse(otherNets[0][1])
+        rootLog = ipAdd(1, p, i)
+    dbg( 2, "Creating network", myNet )
+    topo = InternetTopo(myNet=myNet, rootLog=rootLog)
+    dbg( 3, "Starting on", myNet )
+    net = Mininet(topo=topo, link=TCLink)
+    net.start()
+
+    for host in net.hosts[1:]:
+        host.startCothority()
+
+    # Also set setLogLevel('info') if you want to use this, else
+    # there is no correct reporting on commands.
+    # CLI(net)
+    while not os.path.exists(logdone):
```

Comment:
I'm sure you already searched for it but is there not a way to do like in the `simul` package, to wait until the cothority is done ? Using the utility `wait` for example.